Exercises/NoModules/Chapter 04/Soln4_04.cpp

```diff
@@ -0,0 +1,123 @@
+// Exercise 4-4.
+// As promised, you're to go look for a person "who is over 21, under 35, female, 
+// has a bachelor�s or master�s degree, is unmarried, and who speaks Hindi or Urdu"
+
+#include <iostream>
+#include <cctype>          // For std::tolower() / std::toupper()
+
+enum class AcademicDegree
+{
+  none, associate, bachelor, professional, master, doctor
+};
+
+int main() {
+  int age {};                 // Initialized to 0
+  char gender {};             // Initialized to '\0' (see Chapter 5)
+  AcademicDegree degree {};   // Initialized to AcademicDegree::none
+  bool married {};            // Initialized to false
+  bool speaksHindi {};
+  bool speaksUrdu {};
+
+  std::cout << "What is your age, if I may ask? ";
+  std::cin >> age;
+  
+  if (age < 0 || age > 130)
```

Comment:
This solution doesn't seem to match the question.
The question states:
"People who enter invalid values do not qualify, of course, and should be ruled **out as early as possible (that is, immediately after entering any invalid value;**"

But in your solution, you first collect everything, then at the end you check everytrhing.

_[Edit: on second read, maybe it's ok and I miss interpreted the question. You are just validating input, not yet comparing to what we need.]_

Reply:
Also, why not use unsigned int for age?